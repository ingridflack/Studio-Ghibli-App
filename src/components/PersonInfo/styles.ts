import styled, { css } from "styled-components";

export const Person = styled.div`
  display: flex;
  align-items: center;
`;

export const Info = styled.div`
  ${({ theme }) => css`
    margin-left: 1rem;

    p {
      font-size: ${theme.fontSize.normal};
      display: flex;
      align-items: center;
    }

    span {
      font-size: ${theme.fontSize.medium};
    }

    span + span::before {
      content: "â€¢";
      font-size: ${theme.fontSize.medium};
      margin: 0rem 0.5rem;
    }
  `}
`;

export const PersonImage = styled.img`
  flex-shrink: 0;
  background-color: white;
  border-radius: 50%;
  width: 4rem;
  height: 4rem;
`;
