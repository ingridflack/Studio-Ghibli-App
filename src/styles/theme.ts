const theme = {
  colors: {
    background: "#000",

    gray: {
      white: "#f6f6f6",
      platinum: "#e1e1e6",
      silver: "#a8a8b3",
      lightGray: "#7b8c98",
      dark: "#141414",
    },

    primary: "#e50914",
    secondary: "#b20710",
  },
  fontSize: {
    tiny: "0.65rem",
    small: "0.75rem",
    medium: "0.875rem",
    normal: "1rem",
    large: "1.5rem",
  },
} as const;

export type Theme = typeof theme;

export default theme;
