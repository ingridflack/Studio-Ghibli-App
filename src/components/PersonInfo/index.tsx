import { IPerson } from "../../types";
import * as S from "./styles";

interface PersonInfoProps {
  data: IPerson;
}

const PersonInfo = ({ data }: PersonInfoProps) => {
  const { name, age, gender } = data;

  return (
    <S.Person>
      <S.PersonImage
        src="/assets/user-placeholder.png"
        alt={`${name}'s photo`}
        width="4rem"
        height="4rem"
      />

      <S.Info>
        <p>{name || "Unknown"}</p>
        <span>{(age && `Age: ${age}`) || "Unknown"}</span>
        <span>{gender || "Unknown"}</span>
      </S.Info>
    </S.Person>
  );
};

export default PersonInfo;
