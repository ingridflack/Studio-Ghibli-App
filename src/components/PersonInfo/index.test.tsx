import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import PersonInfo from ".";
import theme from "../../styles/theme";

const PERSON_MOCK = {
  name: "Test name",
  age: "Test age",
  gender: "Test gender",
  id: "Test id",
};

describe("PersonInfo", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <PersonInfo data={PERSON_MOCK} />
      </ThemeProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
