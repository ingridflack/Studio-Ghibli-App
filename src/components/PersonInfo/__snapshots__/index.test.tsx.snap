// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonInfo should match snapshot 1`] = `
<div>
  <div
    class="sc-e5673028-0 fnXhTa"
  >
    <img
      alt="Test name's photo"
      class="sc-e5673028-2 cdofYZ"
      height="4rem"
      src="/assets/user-placeholder.png"
      width="4rem"
    />
    <div
      class="sc-e5673028-1 gsSUVD"
    >
      <p>
        Test name
      </p>
      <span>
        Age: Test age
      </span>
      <span>
        Test gender
      </span>
    </div>
  </div>
</div>
`;
